src/mpi/coll/transports/gentran/gentran_utils.c

```diff
@@ -296,7 +296,12 @@ int MPII_Genutil_progress_hook(int *made_progress)
             DL_DELETE(MPII_coll_queue.head, coll_req);
             MPIR_Request_complete(req);
         }
-        if (++count >= MPIR_CVAR_PROGRESS_MAX_COLLS)
+        if (progress) {
+            count++;
+            if (made_progress)
+                *made_progress = true;
+        }
+        if (count >= MPIR_CVAR_PROGRESS_MAX_COLLS)
```

Comment:
My main issue is this block combined with the initialization of `made_progress` above. It looks awfully convoluted. What if the only time we modify `made_progress` is after the loop? Like this:
```c
if (made_progress) {
    *made_progress = count;
}
```

Reply:
I see. I will fix that.

Reply:
fixed